### PR TITLE
[BD-46] feat: add missing tokens in the Button component

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

- ensure we have a separate design token for every CSS variable hook (example above) used in Button component
image

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: tokens/style-dictionary.js [hunk 1]
```diff
@@ -9,7 +9,13 @@ const cssUtilities = require('./css-utilities');
 const { formattedVariables, fileHeader, sortByReference } = StyleDictionary.formatHelpers;
 
 const colorTransform = (token) => {
-  const { value, modify = [] } = token;
+  const { value, modify = [], original } = token;
+  const reservedColorValues = ['inherit', 'initial', 'revert', 'unset'];
```
Comment (viktorrusakov) [inline tokens/style-dictionary.js]: [clarification] by default style-dictionary replaces such values with color values (e.g. `currentColor` would become `#000000`, see [related issue in style-dictionary](https://github.com/amzn/style-dictionary/issues/459)), this part of code allows us to skip such values without transforming them

File: src/Button/Button.scss [hunk 2]
```diff
@@ -322,61 +379,75 @@ input[type="button"] {
...
+  --pgn-btn-disabled-border-color: #{$btn-brand-disabled-border-color};
   --pgn-btn-active-color: #{$btn-brand-active-color};
   --pgn-btn-active-bg: #{$btn-brand-active-bg};
   --pgn-btn-active-border-color: #{$btn-brand-active-border-color};
+  --pgn-btn-focus-outline-color: #{$btn-brand-focus-outline-color};
+  --pgn-btn-focus-color: #{$btn-brand-focus-color};
+  --pgn-btn-focus-border-color: #{$btn-brand-focus-border-color};
+  --pgn-btn-focus-bg: #{$btn-brand-focus-bg};
 
   box-shadow: $btn-box-shadow;
```
Comment (adamstankiewicz) [inline src/Button/Button.scss:393]: [curious] I see `box-shadow: $btn-box-shadow;` is on many of these `.btn-*` class names, but not all of them. Should it be? Even though I believe the variable is currently essentially set for having no box shadow, if it _did_ have a box shadow, would it be applying equally across all the button variants or only a subset of them?

File: src/Button/_variables.scss [hunk 3]
```diff
@@ -44,34 +46,64 @@ $btn-primary-hover-border-color: var(--pgn-color-btn-hover-border-primary) !defa
 $btn-primary-active-bg: var(--pgn-color-btn-active-bg-primary) !default;
 $btn-primary-active-color: var(--pgn-color-btn-active-text-primary) !default;
 $btn-primary-active-border-color: var(--pgn-color-btn-active-border-primary) !default;
+$btn-primary-focus-focus-outline-color: var(--pgn-color-btn-focus-outline-primary) !default;
```
Comment (adamstankiewicz) [inline src/Button/_variables.scss:49]: [nit/curious] Is this SCSS variable supposed to have the word `focus` back to back? Is this intended to be `$btn-primary-focus-outline-color`?

File: src/Button/Button.scss (hunk 2 above)
Comment (PKulkoRaccoonGang) [reply src/Button/Button.scss:393]: Declaring the `box-shadow: $btn-box-shadow;` is necessary for some buttons on the documentation site. This applies exclusively to the Edxorg theme.

**For example:**
- Openedx theme
image

- Edxorg theme
image

File: src/Button/_variables.scss (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/Button/_variables.scss:49]: Thanks
My bad

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1924&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-01-17+23:06&entry.1917517419=2023-02-15+17:05&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**